### PR TITLE
matrix-appservice-slack: 1.4.0 -> 1.7.0

### PR DESCRIPTION
**Pull request by beardhatcode**
###### Motivation for this change

Bump to the last version I found that compiled... 

if anyone is able to get 1.6.2 working in nix, that would be great. I think this works, but I don't have a working setup yet... (but it is most likely a config error on my side)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by beardhatcode** [comment]
I finally figured out how to update this. And it is now running on my server.

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -1066,6 +1066,12 @@
     githubId = 14111;
     name = "Brandon Dimcheff";
   };
+  beardhatcode = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:1069]: Please put this into a separate commit with the message ``maintainers: add beardhatcode``.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (beardhatcode) [reply maintainers/maintainer-list.nix:1069]: Commit 19b77553918b36b92e702186b84498555fe15bc8 has been reworded to have the subject `maintainers: add beardhatcode` (045e01aae6adaf512b370bd046c5f1ae7ba8909a).

**Comment by expipiplus1** [comment]
@beardhatcode Do you happen to have a nixos module to share for this?

**Comment by beardhatcode** [comment]
I have something like this: https://gist.github.com/beardhatcode/ce57d04da1ec482057a3ac54519359ee/revisions

File: pkgs/servers/matrix-synapse/matrix-appservice-slack/generate-dependencies.sh [hunk 2]
```diff
@@ -1,9 +1,9 @@
 #!/usr/bin/env nix-shell
 #! nix-shell -i bash -p nodePackages.node2nix
 
-# Download package.json and package-lock.json from the v1.4.0 release
-curl https://raw.githubusercontent.com/matrix-org/matrix-appservice-slack/9462a75715ea9afba23b757ec90554f10f457a96/package.json -o package.json
-curl https://raw.githubusercontent.com/matrix-org/matrix-appservice-slack/9462a75715ea9afba23b757ec90554f10f457a96/package-lock.json -o package-lock.json
+# Download package.json and package-lock.json from the v1.7.0 release
+curl https://raw.githubusercontent.com/matrix-org/matrix-appservice-slack/d589aa2e258213ec5ee61ab1d6205c4fb56d116d/package.json -o package.json
+curl https://raw.githubusercontent.com/matrix-org/matrix-appservice-slack/d589aa2e258213ec5ee61ab1d6205c4fb56d116d/package-lock.json -o package-lock.json
```
Comment (SuperSandro2000) [inline pkgs/servers/matrix-synapse/matrix-appservice-slack/generate-dependencies.sh:6]:
```suggestion
curl https://raw.githubusercontent.com/matrix-org/matrix-appservice-slack/1.7.0/package.json -o package.json
curl https://raw.githubusercontent.com/matrix-org/matrix-appservice-slack/1.7.0/package-lock.json -o package-lock.json
```
